tensorboard/webapp/webapp_data_source/tb_feature_flag_data_source_test.ts

```diff
@@ -84,123 +84,197 @@ describe('tb_feature_flag_data_source', () => {
         });
       });
 
-      it('returns enableColorGroup from the query params', () => {
-        getParamsSpy.and.returnValues(
-          new URLSearchParams('enableColorGroup=false'),
-          new URLSearchParams('enableColorGroup='),
-          new URLSearchParams('enableColorGroup=true'),
-          new URLSearchParams('enableColorGroup=foo')
-        );
-        expect(dataSource.getFeatures()).toEqual({enabledColorGroup: false});
-        expect(dataSource.getFeatures()).toEqual({enabledColorGroup: true});
-        expect(dataSource.getFeatures()).toEqual({enabledColorGroup: true});
-        expect(dataSource.getFeatures()).toEqual({enabledColorGroup: true});
+      describe('returns enableColorGroup from the query params', () => {
```

Comment:
To consider for a future PR: Can we greatly reduce the number of tests here? We're effectively testing the same logic over and over again.

Reply:
+1
When I refactored these I was thinking how how screamed to be parameterized.